src/main/java/ust/tad/analysismanager/analysistask/AnalysisTaskService.java

```diff
@@ -5,191 +5,207 @@
+      throw new TransformationProcessNotExistingException(
+          "A transformation process with this id does not exist.");
     }
 
-    public AnalysisTask updateStatusToRunning(AnalysisTask analysisTask, UUID pluginId, AnalysisType analysisType) {
-        analysisTask.setAnalysisType(analysisType);
-        analysisTask.setPluginId(pluginId);
-        analysisTask.setStatus(AnalysisStatus.RUNNING);
-        return analysisTaskRepository.save(analysisTask);
-    }
-
-    public AnalysisTask getTaskByTaskId(UUID taskId) {
-        return analysisTaskRepository.getByTaskId(taskId);
-    }
-
-    public AnalysisTask getOneWaitingAnalysisTask(UUID transformationProcessId) {
-        return analysisTaskRepository.getByStatusAndTransformationProcessId(AnalysisStatus.WAITING, transformationProcessId).get(0);
-    }
-
-    public Boolean areTasksWaiting(UUID transformationProcessId) {
-        return !analysisTaskRepository.getByStatusAndTransformationProcessId(AnalysisStatus.WAITING, transformationProcessId).isEmpty();
-    }
-
-    public Boolean areTasksWaitingOrRunning(UUID transformationProcessId) throws TransformationProcessNotExistingException {
-        if (!analysisTaskRepository.existsByTransformationProcessId(transformationProcessId)) {
-            throw new TransformationProcessNotExistingException("A transformation process with this id does not exist.");
-        }
-
-        return !analysisTaskRepository.getByStatusAndTransformationProcessId(AnalysisStatus.WAITING, transformationProcessId).isEmpty() || !analysisTaskRepository.getByStatusAndTransformationProcessId(AnalysisStatus.RUNNING, transformationProcessId).isEmpty();
-    }
-
-    public Boolean hasNoTaskFailed(UUID transformationProcessId) {
-        return analysisTaskRepository.getByStatusAndTransformationProcessId(AnalysisStatus.FAILED, transformationProcessId).isEmpty();
-    }
-
-    private static class Process {
-        UUID mainTaskId = null;
-        UUID transformationProcessId = null;
-        UUID visualizationTaskId = null;
-        boolean visualize = false;
-        int activeTasks = 0;
-        List<String> commands;
-    }
+    return !analysisTaskRepository
+            .getByStatusAndTransformationProcessId(AnalysisStatus.WAITING, transformationProcessId)
+            .isEmpty()
+        || !analysisTaskRepository
+            .getByStatusAndTransformationProcessId(AnalysisStatus.RUNNING, transformationProcessId)
+            .isEmpty();
+  }
+
+  public Boolean hasNoTaskFailed(UUID transformationProcessId) {
+    return analysisTaskRepository
+        .getByStatusAndTransformationProcessId(AnalysisStatus.FAILED, transformationProcessId)
+        .isEmpty();
+  }
+
+  private static class Process {
+    UUID mainTaskId = null;
+    UUID transformationProcessId = null;
+    UUID visualizationTaskId = null;
+    boolean visualize = false;
+    int activeTasks = 0;
+    List<String> options;
```

Comment:
should comments really be removed here?

Reply:
Commands were only cached because they could contain visualization flags.
As the flags now have to be passed as options, the commands can be removed here.